src/model/Enemy.hs

```diff
@@ -2,130 +2,112 @@
 {-# LANGUAGE InstanceSigs #-}
 module Model.Enemy where
 
-import Model.General (Position, Vector)
-import Model.Shooting (Bullet (Bullet, bulletPos, bulletVector))
+import Model.General (Position, Vector, move)
+import Model.Parameters
+import Model.Shooting (CanShoot, shoot, Bullet (Bullet, bulletPos, bulletVector))
 import Model.Player
 
 -- All five enemy data types, enemy type class.
-data BasicEnemy = BasicEnemy {basicEnemyPos :: Position, basicEnemyCooldown :: Int} 
-data BurstEnemy = BurstEnemy {burstEnemyPos :: Position, burstEnemyCooldown :: Int} 
-data ConeEnemy = ConeEnemy {coneEnemyPos :: Position, coneEnemyCooldown :: Int} 
-data BasicPlayerSeekingEnemy = BasicPlayerSeekingEnemy {basicSeekingPos :: Position, basicSeekingCooldown :: Int} 
-data FastPlayerSeekingEnemy = FastPlayerSeekingEnemy {fastSeekingPos :: Position} 
+data BasicEnemy = BasicEnemy {basicEnemyPos :: Position, basicEnemyCooldown :: Float}
+data BurstEnemy = BurstEnemy {burstEnemyPos :: Position, burstEnemyCooldown :: Float}
+data ConeEnemy = ConeEnemy {coneEnemyPos :: Position, coneEnemyCooldown :: Float}
+data BasicPlayerSeekingEnemy = BasicPlayerSeekingEnemy {basicSeekingPos :: Position, basicSeekingCooldown :: Float}
+data FastPlayerSeekingEnemy = FastPlayerSeekingEnemy {fastSeekingPos :: Position}
 
 
 -- All enemies are instances of this class.
 class Enemy a where 
-    -- Given an enemy and a bullet, return these arguments if the enemy is hit for despawning both. 
-    hit:: a -> Bullet -> Maybe (a, Bullet) 
-    -- Render enemy at a random vertical position.
-    spawn :: a
-    -- Return the position of an Enemy
-    pos :: a -> Position
-
--- Default values, change these later.
-screenWidth :: Int
-screenWidth = 500
-randomY :: Int
-randomY = 300   
-
--- Implementation of method 'hit' for each given enemy. 
-enemyIsHit :: Enemy e => e -> Bullet -> Maybe (e, Bullet)
```

Comment:
Tijdens het maken van de `yMovementSign` functie hieronder bedacht ik me dat je ook gewoon implementaties in de class kan zetten, dus heb dat ook meteen voor `hit` gedaan. Scheelt wat code :)